app/routes.py

```diff
@@ -1 +1,224 @@
-from flask import Blueprint
\ No newline at end of file
+from app import db 
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, abort, make_response, request
+from sqlalchemy import asc, desc
+from datetime import datetime
+import requests
+import os
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+#helper functions
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"{message":f"{cls.__name__} {model_id} invalid"}, 400))
+            
+    model = cls.query.get(model_id)
+    
+    if not model: 
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+
+#route functions 
+
+#POST /tasks
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    if ("title" not in request_body or "description" not in request_body):
+        abort(make_response({"details": "Invalid data"},400))
+
```

Comment:
This is a good attempt at error handling, however, best practice would be to use a try/except block here because we wouldn't have to define all the errors that could happen. For example:

      try:
           new_task = Task.from_dict(request_body)
      except KeyError as error:
           abort(make_response({"details": "Invalid data"}, 400))